backend/src/main/java/de/kaiheinrich/projectplace/controller/ProfileController.java

```diff
@@ -0,0 +1,36 @@
+package de.kaiheinrich.projectplace.controller;
+
+import de.kaiheinrich.projectplace.dto.ProfileDto;
+import de.kaiheinrich.projectplace.model.Profile;
+import de.kaiheinrich.projectplace.service.ProfileService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/profiles")
+public class ProfileController {
+
+    private final ProfileService profileService;
+
+    @Autowired
+    public ProfileController(ProfileService profileService) {
+        this.profileService = profileService;
+    }
+
+    @GetMapping
+    public List<Profile> getProfiles() {
+        return profileService.getProfiles();
+    }
+
+    @PutMapping("{username}")
+    public Profile updateProfile(@RequestBody ProfileDto profileDto, @PathVariable String username) {
+        if(profileService.getProfileByUsername(username).isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
```

Comment:
`profileService.getProfileByUsername(username)` will never return an empty Optional but throw with a status of NOT_FOUND. This looks like a bug. Maybe write tests to assert the correct behaviour?